subcommand/sync-catalog/command.go

```diff
@@ -197,19 +197,12 @@ func (c *Command) Run(args []string) int {
 	}
 
 	// Convert allow/deny lists to sets
-	allowSet := mapset.NewSet()
-	denySet := mapset.NewSet()
+	allowSet := flags.ToSet(c.flagAllowK8sNamespacesList)
+	denySet := flags.ToSet(c.flagDenyK8sNamespacesList)
 	if c.flagK8SSourceNamespace != "" {
 		// For backwards compatibility, if `flagK8SSourceNamespace` is set,
 		// it will be the only allowed namespace
-		allowSet.Add(c.flagK8SSourceNamespace)
-	} else {
-		for _, allow := range c.flagAllowK8sNamespacesList {
-			allowSet.Add(allow)
-		}
-		for _, deny := range c.flagDenyK8sNamespacesList {
-			denySet.Add(deny)
-		}
+		allowSet = mapset.NewSet(c.flagK8SSourceNamespace)
 	}
```

Comment:
This code change is actually slightly different logic because `denySet` could end up with the contents of `c.flagDenyK8sNamespacesList` however this case is only run for older helm versions that weren't passing the `-allow-k8s-ns` or `-deny-k8s-ns` flags at all so this won't matter.